docs/spec_crashreport.rst

```diff
@@ -0,0 +1,218 @@
+===========================
+Specification: Crash report
+===========================
+
+This is a specification for the format for submitting crash reports.
+
+.. contents::
+   :local:
+
+crash report: v0
+================
+
+v0 refers to whatever we had prior to when we moved crash annotations into a
+single JSON-encoded field in `bug 1420363
+<https://bugzilla.mozilla.org/show_bug.cgi?id=1420363>`_. That work landed in
+December 2019 and is in Firefox 73.
+
+
+Submitting a crash report
+-------------------------
+
+Crash reports are submitted by HTTP POST to a URL for a crash ingestion
+collector.
+
+
+Crash report HTTP POST headers
+------------------------------
+
+The content type header of the crash report must be ``multipart/form-data``
+and specify the multipart/form-data boundary.
+
+The content length of the crash report must be set. The value is the length
+of the body.
+
+If the HTTP body is gzip compressed, then the ``Content-Encoding`` must be
+set to ``gzip``.
```

Comment:
nit: You're mentioning this in the following paragraph

Reply:
I tweaked it a bit. This sentence is about headers. The second sentence below is about the body. I'm trying to keep the sections self-contained even if that causes some redundancy in the whole document. Makes it easier for future me to remember all the bits.

Thank you!